### PR TITLE
ci: Skip build/release jobs if only mobile or docs changed

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
@alyssawilk this will require a similar change to the required jobs `release` -> `released` for each of x64/arm64

**Review by alyssawilk — DISMISSED** [review]

File: .azure-pipelines/pipelines.yml [hunk 1]
```diff
@@ -286,11 +286,12 @@ stages:
         AZP_BRANCH: $(Build.SourceBranch)
 
 - stage: linux_x64
-  dependsOn: ["precheck"]
+  dependsOn: ["env", "precheck"]
   # For post-submit builds, continue even if precheck fails
   condition: and(not(canceled()), or(succeeded(), eq(variables['PostSubmit'], true)))
   jobs:
   - job: release
+    condition: and(not(canceled()), succeeded(), ne(stageDependencies.env.repo.outputs['changed.mobileOnly'], 'true'), ne(stageDependencies.env.repo.outputs['changed.docsOnly'], 'true'))
```
Comment (phlax) [inline .azure-pipelines/pipelines.yml:294]: probably we can remove the `not(canceled())` logic here as that is handled by the stage condition

Comment (phlax) [reply .azure-pipelines/pipelines.yml]: i have removed the succeded condition as that is handled above but left the not canceled as we defo dont want it to run in that case

**Review by alyssawilk — DISMISSED** [review]
LGTM though @jpsim or @lizan might actually be able to catch errors =P

File: .azure-pipelines/pipelines.yml [hunk 2]
```diff
@@ -299,12 +300,26 @@ stages:
...
 
+  - job: released
+    dependsOn: ["release"]
+    pool:
+      vmImage: "ubuntu-20.04"
+    # This condition ensures that this (required) job passes if all of
+    # the preceeding jobs either pass or are skipped
+    # adapted from:
+    #   https://learn.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#job-to-job-dependencies-within-one-stage
+    condition: in(dependencies.released.result, 'Succeeded', 'SucceededWithIssues', 'Skipped')
```
Comment (phlax) [inline .azure-pipelines/pipelines.yml:311]:
```suggestion
    condition: in(dependencies.release.result, 'Succeeded', 'SucceededWithIssues', 'Skipped')
```

**Comment by phlax** [comment]
just realizing we need to skip docker also if these are skipped as it uses their artefacts

**Comment by phlax** [comment]
checking through the affected pipelines i think we need to tackle the `docker` and `verify` stages first

**Comment by phlax** [comment]
ive opened #24417 to tackle ^^ first

**Comment by phlax** [comment]
required changes:

- `linux_x64 release` -> `linux_x64 released`
- `linux_arm64 release` -> `linux_arm64 released`
- `macos test` -> `macos tested`
- `windows release` -> `windows released`

remove:

- `windows docker windows2019`
- `windows docker windows2022`

**Pull request merged**